extension/src/server/GradleServer.ts

```diff
@@ -29,51 +30,60 @@ export class GradleServer {
     constructor(
         private readonly opts: ServerOptions,
         private readonly context: vscode.ExtensionContext,
-        private readonly logger: Logger
+        private readonly logger: Logger,
+        private bspProxy: BspProxy
     ) {}
 
     public async start(): Promise<void> {
-        if (isDebuggingServer()) {
-            this.port = 8887;
-            this.fireOnStart();
-        } else {
-            this.port = await getPort();
-            const cwd = this.context.asAbsolutePath("lib");
-            const cmd = path.join(cwd, getGradleServerCommand());
-            const env = await getGradleServerEnv();
-            if (!env) {
+        this.gradleServerPort = await getPort();
+        const cwd = this.context.asAbsolutePath("lib");
+        const cmd = path.join(cwd, getGradleServerCommand());
+        const env = await getGradleServerEnv();
+        const bundleDirectory = this.context.asAbsolutePath("server");
+        if (!env) {
+            await vscode.window.showErrorMessage(NO_JAVA_EXECUTABLE);
+            return;
+        }
+        let startBuildServer = "false";
+        if (redHatJavaInstalled()) {
+            const javaExecPath = getRedHatJavaExecutablePath() || (await getJavaExecutablePath());
+            if (javaExecPath === undefined) {
                 await vscode.window.showErrorMessage(NO_JAVA_EXECUTABLE);
```

Comment:
This will pops up a error message, is it by design?

Reply:
Yes. Design here: 
if redHat.java installed: 
- Find redHat embeded jre --(if not found)--> try our best to find java executable with `jdk>=17`
- if still not found, **showErrorMessage**

If not installed:
- meaning that `build server`&`importer` won't be invoked, but we still need to move forward and start MergeServer

Merge Server will check if java executable exist and decide whether start `build server` or not.